src/fdc/DiskManipulator.cc

```diff
@@ -454,11 +441,27 @@ void DiskManipulator::create(std::span<const TclObject> tokens)
 	}
 }
 
-void DiskManipulator::format(DriveSettings& driveData, MSXBootSectorType bootType)
+void DiskManipulator::format(std::span<const TclObject> tokens)
 {
-	auto partition = getPartition(driveData);
-	DiskImageUtils::format(partition, bootType);
-	driveData.workingDir[driveData.partition] = '/';
+	MSXBootSectorType bootType = MSXBootSectorType::DOS2;
+	std::optional<string> drive;
+	for (const auto& token_ : view::drop(tokens, 2)) {
+		if (auto t = parseBootSectorType(token_.getString())) {
+			bootType = *t;
+		} else if (!drive) {
+			drive = token_.getString();
+		} else {
+			throw CommandException("Incorrect number of parameters");
+		}
+	}
+	if (drive) {
+		auto& driveData = getDriveSettings(*drive);
+		auto partition = getPartition(driveData);
+		DiskImageUtils::format(partition, bootType);
+		driveData.workingDir[driveData.partition] = '/';
+	} else {
+		throw CommandException("Incorrect number of parameters");
+	}
```

Comment:
Very much a detail, coding style:
```
if (!drive) {
	throw CommandException("Incorrect number of parameters");
}
auto& driveData = getDriveSettings(*drive);
auto partition = getPartition(driveData);
DiskImageUtils::format(partition, bootType);
driveData.workingDir[driveData.partition] = '/';
```
This keep the normal code flow non-indented (and better separates it from the error path).
No problem at all if you like your version better.